processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingMethodOptions.java

```diff
@@ -232,6 +245,29 @@ public void applyInheritedOptions(SourceMethod sourceMethod, SourceMethod templa
         }
     }
 
+    private boolean methodsHaveIdenticalSignature(SourceMethod templateMethod, SourceMethod sourceMethod) {
+        return parametersAreOfIdenticalTypeAndOrder( templateMethod, sourceMethod )
+            && resultTypeIsTheSame( templateMethod, sourceMethod );
+    }
+
+    private boolean parametersAreOfIdenticalTypeAndOrder(SourceMethod templateMethod, SourceMethod sourceMethod) {
+        if (templateMethod.getParameters().size() != sourceMethod.getParameters().size()) {
+            return false;
+        }
+        for ( int i = 0; i < templateMethod.getParameters().size(); i++ ) {
+            if (!typeUtils.isSameType( templateMethod.getParameters().get( i ).getType().getTypeMirror(),
```

Comment:
I think that here we can just use `tempalteMethod.getParameter.get( i ).getType().equals( sourceMethod.getParameters().get( i ).getType() )`